lib/core/src/data/function/callback.rs

```diff
@@ -0,0 +1,135 @@
+use crate::prelude::*;
+use std::fmt;
+
+// =====================
+// === Callback Type ===
+// =====================
+
+pub type NoCallback = ();
+
+#[derive(Shrinkwrap)]
+#[shrinkwrap(mutable)]
+pub struct Callback<Func>(pub Func);
+
+impl<Func> Default for Callback<Func> {
+    fn default() -> Self {
+        //        Callback(NOP::nop())
+        unimplemented!()
+    }
+}
+
+impl<Func> fmt::Debug for Callback<Func> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "Callback")
+    }
+}
+
+// ==========================
+// === Callback Interface ===
+// ==========================
+
+pub trait Callback0: 'static {
```

Comment:
Maybe add comment remembering, that there is unstable feature allowing implementing Fn trait (e.g. `impl Fn(i32) for Foo`) and we could use them instead of own CallbackN traits